cli/azd/pkg/commands/pipeline/pipeline.go

```diff
@@ -58,14 +64,20 @@ type ScmProvider interface {
 		remoteName string,
 		branchName string,
 		console input.Console) (bool, error)
+	//Hook function to allow SCM providers to handle scenarios after the git push is complete
+	postGitPush(ctx context.Context,
+		gitRepo *gitRepositoryDetails,
+		remoteName string,
+		branchName string,
+		console input.Console) error
 }
 
 // CiProvider defines the base behavior for a continuous integration provider.
 type CiProvider interface {
 	// compose the behavior from subareaProvider
 	subareaProvider
 	// configurePipeline set up or create the CI pipeline.
-	configurePipeline(ctx context.Context) error
+	configurePipeline(ctx context.Context, repoDetails *gitRepositoryDetails, provisioningProvider provisioning.Options) error
```

Comment:
maybe instead of `provisioningProvider` name this argument `options` or something, since you pass in options, not a provider?

Reply:
https://github.com/Azure/azure-dev/issues/756